fathom/tests/source_tests.rs

```diff
@@ -138,73 +133,63 @@ pub fn find_source_files(root: impl AsRef<Path>) -> impl Iterator<Item = PathBuf
         .map(|entry| entry.into_path())
 }
 
-fn extract_module_test(path: PathBuf) -> libtest_mimic::Test<TestData> {
-    libtest_mimic::Test {
-        name: path.display().to_string(),
-        kind: String::new(),
-        is_ignored: false,
-        is_bench: false,
-        data: TestData {
-            input_file: path,
-            mode: TestMode::Module,
-        },
-    }
+fn extract_module_test(path: PathBuf) -> libtest_mimic::Trial {
+    extract_test(path, TestMode::Module)
 }
 
-fn extract_term_test(path: PathBuf) -> libtest_mimic::Test<TestData> {
-    libtest_mimic::Test {
-        name: path.display().to_string(),
-        kind: String::new(),
-        is_ignored: false,
-        is_bench: false,
-        data: TestData {
-            input_file: path,
-            mode: TestMode::Term,
-        },
-    }
+fn extract_term_test(path: PathBuf) -> libtest_mimic::Trial {
+    extract_test(path, TestMode::Term)
 }
 
-fn run_test(test: &libtest_mimic::Test<TestData>) -> libtest_mimic::Outcome {
-    let mut failures = Vec::new();
+fn extract_test(input_file: PathBuf, default_mode: TestMode) -> libtest_mimic::Trial {
```

Comment:
It would be nicer to instead have:

```rust
fn extract_test(input_file: PathBuf, default_mode: TestMode) -> Result<(), libtest_mimic::Failure> {
```

But libtest-mimic [no longer lets you dynamically ignore tests](https://github.com/LukasKalbertodt/libtest-mimic/issues/19) (as of v0.5). Hopefully this changes in the future!